src/structures/Message.js

```diff
@@ -179,54 +181,73 @@ class Message extends Base {
      * A manager of the reactions belonging to this message
      * @type {ReactionManager}
      */
-    this.reactions = new ReactionManager(this);
+    this.reactions = partial && this.reactions ? this.reactions : new ReactionManager(this);
     if (data.reactions?.length > 0) {
       for (const reaction of data.reactions) {
         this.reactions.add(reaction);
       }
     }
 
-    /**
-     * All valid mentions that the message contains
-     * @type {MessageMentions}
-     */
-    this.mentions = new Mentions(
-      this,
-      data.mentions,
-      data.mention_roles,
-      data.mention_everyone,
-      data.mention_channels,
-      data.referenced_message?.author,
-    );
+    if (partial) {
+      this.mentions = new Mentions(
+        this,
+        data.mentions ?? this.mentions.users,
```

Comment:
Can't this now throw an error for an uncached message (with partials enabled)? Since `this.mentions` is never defined

Reply:
No, the only way this could cause errors is if the _patch call in the constructor sets partial to true, so don't do that!

Reply:
But the MessageUpdateAction literally may create a new Message object and call _patch with partial=true...so, are you sure this isn't an issue?

Reply:
You can't call _patch(data, true) on an uninstantiated message. In creating the new Message object via _add, it calls _patch(data), not partial, then patches as partial later.